src/hotspot/share/opto/vectornode.cpp

```diff
@@ -2194,6 +2194,65 @@ bool MulVLNode::has_uint_inputs() const {
          has_vector_elements_fit_uint(in(2));
 }
 
+static Node* UMinMaxV_Ideal(Node* n, PhaseGVN* phase, bool can_reshape) {
+  int vopc = n->Opcode();
+  assert(vopc == Op_UMinV || vopc == Op_UMaxV, "Unexpected opcode");
+
+  Node* umin = nullptr;
+  Node* umax = nullptr;
+  int lopc = n->in(1)->Opcode();
+  int ropc = n->in(2)->Opcode();
+
+  if (lopc == Op_UMinV && ropc == Op_UMaxV) {
+    umin = n->in(1);
+    umax = n->in(2);
+  }
+  else if (lopc == Op_UMaxV && ropc == Op_UMinV) {
+    umin = n->in(2);
+    umax = n->in(1);
+  }
```

Comment:
```suggestion
  } else {
    // Either both Min or Max.
    return nullptr;
  }
```
That way, you don't need the check below:
`umin != nullptr && umax != nullptr`.

Reply:
We still want to execute VectorNode::Ideal if new transforms are not applicable.